File: ruby/test/lib/yamlscript_test.rb
require 'minitest/autorun'

require_relative '../../lib/yamlscript'

class YAMLScriptTest < Minitest::Test
  def setup
    @ys = YAMLScript.new
  end

  def test_load_yaml_data
    data = @ys.load('a: 1')

    assert_equal data['a'], 1
  end
end

File: scala/test/YAMLScriptTest.scala
//> using scala 3.7.3

package org.yamlscript

object YAMLScriptTest:
  def main(args: Array[String]): Unit =
    val data = YAMLScript.load("!ys-0:\ntest:: inc(41)")
    assert(data("test").num.toInt == 42)
    println("ok - load ys code")
